### ISSUE TITLE
No field validation if EventPreSaveFilter fails

### ISSUE THREAD

**Issue by andrewminton** [symphonycms/symphony-2#2500]
Is it normal for the post values to be missing if you submit a form and the filters fail?

I'm trying to present the full list of required fields as well as notify the user that a filter has failed but all I get is the filter error message and optional fields returned in the event xml. i.e. missing required field post values do not appear. 

If I comment out the '$context['messages'][]' part of my extension, these post values appear all the time.

How is the core dealing with filter messages?

**Comment by andrewminton** [comment]
I'm looking specifically at the EventPreSaveFilter delegate at the moment.

**Comment by michael-e** [comment]
I built similar filters yesterday, I will do some tests and come back!

**Comment by michael-e** [comment]
No, I can't confirm this. If such a filter fails, you should still have POST values in your event XML.

How does your array look like? It should be similar to this:

```php
$context['messages'][] = array(
    'my-custom-filter',
    false,
    __('My custom filter has failed, so your are lost.')
);
```

**Comment by michael-e** [comment]
But, of course (and that's the culprit), you won't have validation on the fields (because the filter returning `false` stops the process, unfortunately).

**Comment by michael-e** [comment]
> i.e. missing required field post values

Ah, by that you mean validation? OK, so I can confirm. There won't be any field validation (see above).

But I am afraid that this is not a bug (but intended behaviour).

**Comment by michael-e** [comment]
I assume that your filter is doing some sort of "advanced validation", and you would like to combine this with field validation. (That is what I tried yesterday.) This is another interesting use case for additional (or pimped) delegates. :-) We thought about these here as well: https://github.com/symphonycms/symphony-2/issues/2369#issuecomment-142950485

**Comment by michael-e** [comment]
BTW, Happy birthday, issue tracker! You are 2500 isues old!

**Comment by andrewminton** [comment]
Haha! Happy Birthday :)

What you have mentioned is correct yes. I'm rebuilding the recaptcha extension with evil Googles method and when the filter fails, there is no validation of fields along with the filter fail message.

I would have thought, we'd still want to report validation errors, but I guess the filter trumps this and stops it before it gets that far.

I can live with it, just curious to know if this was the norm.

Thanks for checking @michael-e

**Comment by michael-e** [comment]
I tend to leave this issue open nevertheless, because it shows an interesting aspect of the delegate problem. I would be great to find a (backwards-compatible) way to modify and/or add delegates which can solve this issue and show a way to deal with #2369 as well, i.e. have delegates for backend and frontend which allow to validate and manipulate field data in the same way. (I mentioned in https://github.com/symphonycms/symphony-2/issues/2369#issuecomment-142950485 that this is not the case at the moment.)

Maybe @nitriques comes up with a cool suggestion? (I don't have time at the moment, unfortuantely.)

**Comment by andrewminton** [comment]
Thanks for info :)

**Comment by nitriques** [comment]
So can we close if this is not an issue ? Or we should rename this "epic" issue :) ?

**Comment by michael-e** [comment]
Renamed and closed.

**Comment by nitriques** [comment]
Thanks Michael!

**Comment by andrewminton** [comment]
If the move you're suggesting exposes post values even if filter fails. That's a winner all round.

**Comment by michael-e** [comment]
You mean validation errors? Well, with this fix validation would be first, so you would get the field errors. Only if validation succeeds (which means no field errors occur), the filter is executed.

**Comment by andrewminton** [comment]
Ok good. and if filter fails? would the event xml still retain the posted values data? I guess it would right?

**Comment by michael-e** [comment]
I thinks so, depending on how `processPreSaveFilters` is cleaned up. It includes a fix for #2179, @brendo will hopefully know how to deal with that. :-)
